packages/rum-core/src/domain/waitIdlePage.ts

```diff
@@ -80,7 +81,7 @@ export function doWaitIdlePage(
     const lastChangeTime = timeStampNow()
     if (!isBusy) {
       idleTimeoutId = setTimeout(
-        monitor(() => complete({ hadActivity: true, duration: elapsed(startTime, lastChangeTime) })),
+        monitor(() => complete({ hadActivity: true, duration: elapsed(startClocks.timeStamp, lastChangeTime) })),
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Instead of passing the start clocks around, I would change the `IdlePageEvent` so instead of a `duration` it has an `end` timestamp, and use this `end` to compute the final duration wherever `waitIdlePage` is used.

Reply:
+1 good suggestion

Reply:
hum, duration is also used internally to handle cases with max duration:
https://github.com/DataDog/browser-sdk/blob/2ee259897465789c05f404e8a57fa98a4183c73d/packages/rum-core/src/domain/waitIdlePage.ts#L71-L76
wdyt?

Reply:
What would be the issue with
```ts
monitor(() => complete({ hadActivity: true, end: timeStampNow() })), 
```
?

Reply:
I don't really see the relation with maxDuration.

Reply:
nevermind, I get confused, it should be ok